src/Workspaces/Core/Portable/LanguageServices/SyntaxFactsService/AbstractSyntaxFactsService.cs

```diff
@@ -158,8 +158,12 @@ protected AbstractSyntaxFactsService()
                     Matcher.Repeat(singleBlankLine));
         }
 
-        public abstract bool IsWhitespaceTrivia(SyntaxTrivia trivia);
-        public abstract bool IsEndOfLineTrivia(SyntaxTrivia trivia);
+        private bool IsWhitespaceTrivia(SyntaxTrivia trivia)
+            => SyntaxKinds.WhitespaceTrivia == trivia.RawKind;
+
+        private bool IsEndOfLineTrivia(SyntaxTrivia trivia)
+            => SyntaxKinds.EndOfLineTrivia == trivia.RawKind;
```

Comment:
couldn't move due to some weird inheritance shenanigans.  Will fixup as more cleanup happens.  but there is already a ton of goodness in this PR.